prov/rxd/src/rxd_cq.c

```diff
@@ -160,12 +160,43 @@ static int rxd_cq_write_tagged_signal(struct rxd_cq *cq,
 
 void rxd_rx_entry_free(struct rxd_ep *ep, struct rxd_x_entry *rx_entry)
 {
-	rx_entry->state = RXD_FREE;
-	rx_entry->key = ~0;
+	rx_entry->rx_id = ~0;
+	rx_entry->msg_id = ~0;
 	dlist_remove(&rx_entry->entry);
 	freestack_push(ep->rx_fs, rx_entry);
 }
 
+
+static void rxd_remove_rx_pkt(struct rxd_ep *ep, struct rxd_pkt_entry *pkt_entry)
+{
+	struct rxd_pkt_entry *pkt_entry_head;
+	struct slist_entry *item, *prev;
+
+	pkt_entry_head = container_of((&ep->rx_pkt_list)->head,
+				      struct rxd_pkt_entry, s_entry);
+
+	if (pkt_entry_head == pkt_entry) {
+		slist_remove_head(&ep->rx_pkt_list);
+		return;
+	}
+
+	FI_WARN(&rxd_prov, FI_LOG_EP_CTRL,
+		"matched to incorrect rx\n");
+	slist_foreach(&ep->rx_pkt_list, item, prev) {
+		pkt_entry_head = container_of(item, struct rxd_pkt_entry,
+					      s_entry);
+		if (pkt_entry_head == pkt_entry)
+			break;
+	}
+	slist_remove(&ep->rx_pkt_list, item, prev);
+}
```

Comment:
There's an slist_remove_first_match() function that I think can be used here instead.  Note that in the above code, if slist_foreach doesn't find a match we remove... something?